From: Steven Rostedt <srostedt@redhat.com>
Subject: ftrace: fix if define to prove locking

The preprocessor condition in fork.c incorrectly checks against LOCKDEP,
when it should check against PROVE_LOCKING.

Signed-off-by: Steven Rostedt <srostedt@redhat.com>
---
 kernel/fork.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.24.7-rt27/kernel/fork.c
===================================================================
--- linux-2.6.24.7-rt27.orig/kernel/fork.c	2009-02-08 00:04:25.000000000 -0500
+++ linux-2.6.24.7-rt27/kernel/fork.c	2009-02-08 00:04:31.000000000 -0500
@@ -1051,7 +1051,7 @@ static struct task_struct *copy_process(
 
 	rt_mutex_init_task(p);
 
-#if defined(CONFIG_TRACE_IRQFLAGS) && defined(CONFIG_LOCKDEP)
+#if defined(CONFIG_TRACE_IRQFLAGS) && defined(CONFIG_PROVE_LOCKING)
 	DEBUG_LOCKS_WARN_ON(!p->hardirqs_enabled);
 	DEBUG_LOCKS_WARN_ON(!p->softirqs_enabled);
 #endif
